sympy/simplify/tests/test_ratsimp.py

```diff
@@ -74,5 +74,5 @@ def test_ratsimpmodprime():
         y/2
 
     a = (x**5 + 2*x**4 + 2*x**3 + 2*x**2 + x + 2/x + x**(-2))
-    assert ratsimpmodprime(a, [x + 1], domain=GF(2)) == 1
-    assert ratsimpmodprime(a, [x + 1], domain=GF(3)) == -1
+    # assert ratsimpmodprime(a, [x + 1], domain=GF(2)) == 1
+    # assert ratsimpmodprime(a, [x + 1], domain=GF(3)) == -1
```

Comment:
These examples highlight a problem with disallowing any unification between `GF(p)` and `ZZ`:
```python
In [1]: p = Poly(1, x, modulus=2)

In [2]: p
Out[2]: Poly(1, x, modulus=2)

In [3]: p * 2
...
UnificationFailed: Cannot unify GF(2) with ZZ
```
There are probably ways to work around this but at least it shows why being able to unify `GF(p)` and `ZZ` makes sense in some situations. Here the problem is that the `2` is coerced into `Poly(2, x, domain=ZZ)` before the product is computed. Then the product tries to unify the two `Poly` and fails if `GF(2)` cannot unify with `ZZ`.

Reply:
In this example I don't think it matters which way the unification happens as long as they do unify to something.

Reply:
I fixed this by special casing `Poly * int/Integer`.